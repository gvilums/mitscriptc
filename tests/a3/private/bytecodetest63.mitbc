function
{
	functions =
	[
		function
		{
			functions = [],
			constants = [],
			parameter_count = 1,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
			]
		},
		function
		{
			functions = [],
			constants = [],
			parameter_count = 0,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
			]
		},
		function
		{
			functions = [],
			constants = [],
			parameter_count = 1,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
			]
		},
		function
		{
			functions =
			[
				function
				{
					functions = [],
					constants = [None],
					parameter_count = 1,
					local_vars = [leftNode],
					local_ref_vars = [],
					free_vars = [this],
					names = [left],
					instructions = 
					[
						push_ref	0
						load_ref
						load_local	0
						field_store	0
						load_const	0
						return
					]
				},
				function
				{
					functions = [],
					constants = [None],
					parameter_count = 1,
					local_vars = [rightNode],
					local_ref_vars = [],
					free_vars = [this],
					names = [right],
					instructions = 
					[
						push_ref	0
						load_ref
						load_local	0
						field_store	0
						load_const	0
						return
					]
				},
				function
				{
					functions = [],
					constants = [],
					parameter_count = 0,
					local_vars = [],
					local_ref_vars = [],
					free_vars = [this],
					names = [left],
					instructions = 
					[
						push_ref	0
						load_ref
						field_load	0
						return
					]
				},
				function
				{
					functions = [],
					constants = [],
					parameter_count = 0,
					local_vars = [],
					local_ref_vars = [],
					free_vars = [this],
					names = [right],
					instructions = 
					[
						push_ref	0
						load_ref
						field_load	0
						return
					]
				}
			],
			constants = [None],
			parameter_count = 0,
			local_vars = [this],
			local_ref_vars = [this],
			free_vars = [],
			names = [left, right, setLeft, setRight, getLeft, getRight],
			instructions = 
			[
				load_const	0
				store_local	0
				alloc_record
				dup
				dup
				dup
				dup
				dup
				dup
				load_const	0
				field_store	0
				load_const	0
				field_store	1
				load_func	0
				push_ref	0
				alloc_closure	1
				field_store	2
				load_func	1
				push_ref	0
				alloc_closure	1
				field_store	3
				load_func	2
				push_ref	0
				alloc_closure	1
				field_store	4
				load_func	3
				push_ref	0
				alloc_closure	1
				field_store	5
				store_local	0
				load_local	0
				return
			]
		},
		function
		{
			functions = [],
			constants = [None, 0, 1],
			parameter_count = 1,
			local_vars = [depth, newNode],
			local_ref_vars = [],
			free_vars = [],
			names = [Node, setLeft, maketree, setRight],
			instructions = 
			[
				load_const	0
				store_local	1
				load_local	0
				load_const	1
				swap
				geq
				not
				if	7
				load_global	0
				call	0
				store_local	1
				load_local	1
				return
				goto	24
				load_global	0
				call	0
				store_local	1
				load_local	1
				field_load	1
				load_global	2
				load_local	0
				load_const	2
				sub
				call	1
				call	1
				pop
				load_local	1
				field_load	3
				load_global	2
				load_local	0
				load_const	2
				sub
				call	1
				call	1
				pop
				load_local	1
				return
				load_const	0
				return
			]
		}
	],
	constants = [14, "Garbage collector bench5.mit", "Stretching memory with a binary tree of depth ", None],
	parameter_count = 0,
	local_vars = [],
	local_ref_vars = [],
	free_vars = [],
	names = [kStretchTreeDepth, Node, maketree, temp, print, input, intcast],
	instructions = 
	[
		load_func	0
		alloc_closure	0
		store_global	4
		load_func	1
		alloc_closure	0
		store_global	5
		load_func	2
		alloc_closure	0
		store_global	6
		load_const	0
		store_global	0
		load_func	3
		alloc_closure	0
		store_global	1
		load_func	4
		alloc_closure	0
		store_global	2
		load_global	4
		load_const	1
		call	1
		pop
		load_global	4
		load_const	2
		load_global	0
		add
		call	1
		pop
		load_global	2
		load_global	0
		call	1
		store_global	3
		load_const	3
		store_global	3
	]
}