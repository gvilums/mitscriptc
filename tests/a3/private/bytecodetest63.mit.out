Garbage collector bench5.mit
Stretching memory with a binary tree of depth 14
